### ISSUE TITLE
Dependency default URL (CDN or selfhosted)

### ISSUE THREAD

**Issue by stp-ip** [okkur/syna#86]
**Is this a BUG REPORT or FEATURE REQUEST?**:
feature

**What happened**:
Currently we use self hosted dependencies for bootstrap, fontawesome etc.

**What you expected to happen**:
Discuss our general approach and make switching possible.

**Anything else we need to know?**:
Make it possible to offload bigger libraries to their respective CDN (bootstrap, jquery, fontawesome).

**Comment by mpourismaiel** [comment]
If we introduce webpack the final build could contain all the files bundled nicely together and chunked if needed so particular scripts are only loaded in certain fragments. If that is not nice we can use CDNs along our actual code.

**Comment by stp-ip** [comment]
So let's start with selfhosted and webpack created one css and js file, which doesn't include fragment code. Aka contact css moves to fragment.

Should we also use webpack for the fragment css and js files?

Later we can add code to use a CDN or just leave it to the user to host it differently via custom css/js. So start with what works for us now and wait for what the actual CDN methods could be using user input. Most probably push their site to a CDN enabled host anyway so probably no immediate need to offload to a CDN besides faster load times (bootstrap CDN being cached already etc.)

Does that sound reasonable?

**Comment by stp-ip** [comment]
We have the following deps:
* jquery
* bootstrap
* newly colored bootstrap
* fontawesome
* syna specific stuff

Moving all into one file seems a bit wrong especially with http/2 sending in parallel anyway.

Also, if we ever move to enabling CDN usage or per script overwrites it might be simpled, if these are split from the get go.

The only thing that seems to need to be bundled is boostrap and the color overwrite. All others could be separate files.

You have more experience with webpack and css/js in general so I'm happy to go with your suggestion.

**Comment by stp-ip** [comment]
OK we could split this into 3 stages:
1. webpack generated separate self hosted files + self hosted per fragment files
2. Add switch/config to use dependency CDNs (bootstrap cdn, jquery cdn, fontawseome cdn etc) + syna and fragment being self hosted
3. Option to disable scripts and add custom ones via custom css/js in config.toml <-- config or let them overwrite header.html or so? (custom script support for fragments seems out of scope)

**Comment by stp-ip** [comment]
Considering that we decided on self-hosted deps. https://github.com/okkur/syna/pull/102 introduced a regression and is now relying on an external url.

**Comment by stp-ip** [comment]
Split it into multiple issues. To be more actionable.
